app/src/main/java/de/westnordost/streetcomplete/map/VectorTileProvider.kt

```diff
@@ -1,13 +1,18 @@
 package de.westnordost.streetcomplete.map
 
-abstract class VectorTileProvider(
+data class VectorTileProvider(
+    val sceneFilePath: String,
+    val baseTileSource: TileSource,
+    val rasterLayerSource: TileSource,
```

Comment:
maybe it would be easier if this was simply a list of tile sources, so then it is not necessary to discriminate between "baseTileSource" and "rasterLayerSource" everywhere.

On the other hand, if you plan to only . But as said, maybe first look at what is the actual additional download size of always also downloading the satellite imagery. If it is not much more, it is maybe not worth it to **not** always also download the satellite imagery.

Reply:
This way forces me to discriminate, but the other way forces me to use magic numbers to know which source is which; to me it's much clearer that the base tile source is always used and the raster source is sometimes used. 

I imagined that if we add other renderings in the future, the base layer is probably still used and it's just the visual layer that changes, so I was imagining that in this case, the `rasterLayerSource` would be converted to an array. But while I read enough of the tangram docs and skimmed our yaml to get a sense of what a scene is, I didn't learn the details yet, so maybe this is the wrong abstraction.

I think I should probably just rename `raster` to `aerial` here, and I should add parts of the paragraph above to the commit message.

Reply:
> This way forces me to discriminate, but the other way forces me to use magic numbers to know which source is which

Why do you need to know which is which?

Reply:
Because the base layer is always downloaded and the aerial layer isn't (or if it is, different cache rules will be used).

Reply:
I suppose it would be possible that we didn't need to know which is which, if we stick more information into the `VectorTileProvider` (and rename it to `MapScene` as you suggested earlier). E.g. this info should be stored in the `MapScene`:

https://github.com/streetcomplete/StreetComplete/blob/0d70f5261e1e8c198f24671a94b23eb0a68c626e/app/src/main/java/de/westnordost/streetcomplete/map/components/SceneMapComponent.kt#L82-L86

So then we'd have 3 different map scenes — light, dark, and aerial. Light and dark would each have only one TileSource, and Aerial would have two. All sources would always be downloaded unconditionally. Cache rules would be stored in the data source. This has better separation of concerns, and will scale much better if additional scenes are ever added. So *for the moment* I'm convinced this is a better architecture.

On the other hand, it's also an additional layer of abstraction, which cements this interface. So, I'm leaning towards staying with the baseLayer / aerialLayer split *for now*. Then, later, once the feature has been released for a bit and any related feature requests have come in, we can see if this interface still makes sense, and I'll send a refactoring PR if it does.

Reply:
Added a longer explanation to this effect in the first commit